libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -778,6 +922,29 @@ AP_GPS::broadcast_first_configuration_failure_reason(void) const {
     }
 }
 
+// pre-arm check that all GPSs are close to each other.  farthest distance between GPSs (in meters) is returned
+bool AP_GPS::all_consistent(float &distance) const
+{
+    // return true immediately if only one valid receiver
+    if (num_instances <= 1 ||
+        drivers[0] == nullptr || _type[0] == GPS_TYPE_NONE ||
+        drivers[1] == nullptr || _type[1] == GPS_TYPE_NONE) {
+        distance = 0;
+        return true;
+    }
+
+    // calculate distance
+    distance = location_3d_diff_NED(state[0].location, state[1].location).length();
+    // success if distance is within 50m
+    return (distance < 50);
+}
+
+// pre-arm check of GPS blending.  True means healthy or that blending is not being used
+bool AP_GPS::blend_healthy() const
+{
+    return (_blend_health_counter < 50);
```

Comment:
I think we discussed the naming of the blend_healthy() above so I won't respond to that here.
Thanks for the find on the _blend_health_counter.  That's now set to zero if _auto_switch != 2 (i.e. if the user hasn't requested blending).